File: VB/WindowsApplication1/Data/ParentItemClass.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq

Namespace WindowsApplication1
    Public Class ParentItemClass
        Public Sub New(ByVal name As String, ByVal id As Integer)
            ItemName = name
            ItemID = id
        End Sub

        Public Property ItemName As String
        Public Property ItemID As Integer
        Public Property DetailItems As List(Of DetailItemClass) = New List(Of DetailItemClass)()
    End Class
End Namespace

File: CS/WindowsApplication1/Data/ParentItemClass.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace WindowsApplication1 {
    public class ParentItemClass {
        public ParentItemClass(string name, int id) {
            ItemName = name;
            ItemID = id;
        }
        public string ItemName { get; set; }
        public int ItemID { get; set; }
        public List<DetailItemClass> DetailItems { get; set; } = new List<DetailItemClass>();
    }
}
